### PR TITLE
Add balance data to user_command_status

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR adds balances to the status for user commands. The archive node side is a TODO, see the comment in `archive_lib/processor.ml`

Note that this has dummy values for coinbases and fee-transfers, which aren't surfaced in the archive node. These are slightly more complicated, and I've found (yet another)  bug in the fee-transfers' undo, so I am going to update them separately.

Checklist:

- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

**Review by bkase — APPROVED** [review]

File: src/app/archive/archive_lib/processor.ml [hunk 1]
```diff
@@ -269,32 +269,49 @@ module User_command = struct
...
             ( "failed"
             , Some (User_command_status.Failure.to_string failure)
             , None
             , None
-            , None )
+            , None
+            , balances_to_int64s balances )
       in
+      (* TODO: Record these with the transaction *)
+      ignore (fee_payer_balance, source_balance, receiver_balance) ;
```
Comment (omerzach) [inline src/app/archive/archive_lib/processor.ml:314]: Just for my education, this is just here to suppress a compiler warning?

**Review by omerzach — APPROVED** [review]
This is great, thanks.

One question though: The RFC (https://github.com/MinaProtocol/mina/pull/6940/files) used "sender" rather than "source"; is source more appropriate to use throughout for this?

File: src/app/archive/archive_lib/processor.ml (hunk 1 above)
Comment (mrmr1993) [reply src/app/archive/archive_lib/processor.ml:314]: Yea, spot on.

**Comment by mrmr1993** [comment]
> The RFC (https://github.com/MinaProtocol/mina/pull/6940/files) used "sender" rather than "source"; is source more appropriate to use throughout for this.

Sorry, that was my sloppiness on the RFC. We should use `source` to be consistent with the rest of the codebase.

**Pull request merged**